evm-ds/src/evm_server_run.rs

```diff
@@ -0,0 +1,330 @@
+    node_continuation: Option<EvmProto::Continuation>,
+    continuations: Arc<Mutex<Continuations>>,
+    enable_cps: bool,
+) -> Result<String> {
+    // We must spawn a separate blocking task (on a blocking thread), because by default a JSONRPC
+    // method runs as a non-blocking thread under a tokio runtime, and creating a new runtime
+    // cannot be done. And we'll need a new runtime that we can safely drop on a handled
+    // panic. (Using the parent runtime and dropping on stack unwind will mess up the parent runtime).
+    tokio::task::spawn_blocking(move || {
+        debug!(
+            "Running EVM: origin: {:?} address: {:?} gas: {:?} value: {:?}  extras: {:?}, estimate: {:?}, cps: {:?}",
+            backend.origin, address, gas_limit, apparent_value,
+            backend.extras, estimate, enable_cps);
+        let code = Rc::new(code);
+        let data = Rc::new(data);
+        // TODO: handle call_l64_after_gas problem: https://zilliqa-jira.atlassian.net/browse/ZIL-5012
+        let config = evm::Config { estimate, call_l64_after_gas: false, ..evm::Config::london()};
+        let context = evm::Context {
+            address,
+            caller: backend.origin,
+            apparent_value,
+        };
+        let mut runtime: Runtime;
+        let gas_limit = gas_limit * gas_scaling_factor;
+        let metadata = StackSubstateMetadata::new(gas_limit, &config);
+        let state;
+
+        let mut feedback_continuation:Option<EvmProto::Continuation> = None;
+        // Check if evm should resume from the point it stopped
+        if let Some(continuation) = node_continuation {
+            let recorded_cont = continuations.lock().unwrap().get_contination(continuation.get_id().into());
+            if let None = recorded_cont {
+                let result = handle_panic(vec![], gas_limit);
+                return Ok(base64::encode(result.write_to_bytes().unwrap()));
+            }
+            let recorded_cont = recorded_cont.unwrap();
+            let machine = Machine::create_from_state(Rc::new(recorded_cont.code), Rc::new(recorded_cont.data),
+                                                              recorded_cont.position, recorded_cont.return_range, recorded_cont.valids,
+                                                              recorded_cont.memory, recorded_cont.stack);
+            runtime = Runtime::new_from_state(machine, context, &config);
+            let memory_substate = MemoryStackSubstate::from_state(metadata, recorded_cont.logs, recorded_cont.accounts,
+                recorded_cont.storages, recorded_cont.deletes);
+            state = MemoryStackState::new_with_substate(memory_substate, &backend);
+            feedback_continuation = Some(continuation);
+        }
+        else {
+            runtime = evm::Runtime::new(code.clone(), data.clone(), context, &config);
+            state = MemoryStackState::new(metadata, &backend);
+        }
+        // Scale the gas limit.
```

Comment:
In Rust, you are encouraged to minimize assingment in favor of initialization to minimize mutable state, e.g.

```suggestion
        let (mut runtime, state, feedback_continuation) =
        if let Some(ref continuation) = node_continuation {
            // Evm should resume from the point it stopped
            let recorded_cont = continuations.lock().unwrap().get_contination(continuation.get_id().into());
            if recorded_cont.is_none() {
                let result = handle_panic(vec![], gas_limit);
                return Ok(base64::encode(result.write_to_bytes().unwrap()));
            }
            let recorded_cont = recorded_cont.unwrap();
            let machine = Machine::create_from_state(Rc::new(recorded_cont.code), Rc::new(recorded_cont.data),
                                                              recorded_cont.position, recorded_cont.return_range, recorded_cont.valids,
                                                     recorded_cont.memory, recorded_cont.stack);
            let memory_substate = MemoryStackSubstate::from_state(metadata, recorded_cont.logs, recorded_cont.accounts,
                recorded_cont.storages, recorded_cont.deletes);
            (Runtime::new_from_state(machine, context, &config),
             MemoryStackState::new_with_substate(memory_substate, &backend),
             node_continuation)
        }  else {
            (Runtime::new(code.clone(), data.clone(), context, &config),
             MemoryStackState::new(metadata, &backend),
             None)
        };

```